azure/packages/azure-client/src/test/TestDataObject.ts

```diff
@@ -21,3 +23,64 @@ export class TestDataObject extends DataObject {
         super(props);
     }
 }
+
+export class TestDataObject2 extends DataObject {
+    private _counter: SharedCounter | undefined;
+
+    constructor(props: IDataObjectProps) {
+        super(props);
+    }
+
+    /**
+     * Do setup work here
+     */
+    protected async initializingFirstTime(): Promise<void> {
+        const counter = SharedCounter.create(this.runtime);
+        this.root.set("counter-key", counter.handle);
+    }
+
+    protected async hasInitialized(): Promise<void> {
+        const counterHandle = this.root.get<IFluidHandle<SharedCounter>>("counter-key");
+        this._counter = await counterHandle?.get();
+    }
+
+    public static get Name(): string {
+        return "@fluid-example/test-data-object-2";
+    }
+
+    public static readonly factory = new DataObjectFactory(
+        TestDataObject2.Name,
+        TestDataObject2,
+        [SharedCounter.getFactory()],
+        {},
+    );
+
+    public increment(): void {
+        this.counter.increment(1);
+    }
+
+    public reset(): void {
+        this.counter.increment(1);
+    }
```

Comment:
Counter doesn't support reset, should this method exist?

Reply:
This should not be here. This whole file needed a cleanup.